### DISCUSSION TITLE
Image Read Failed

### DISCUSSION THREAD

**Discussion by programmerarunkumar** [ImageMagick/ImageMagick#2925] [Help]
Hi Im using MagickWand to resize the image 

 MagickReadImage(MagickWand *wand,const char *filename) return MagickFalse

Is there any general reason for returning MagickFalse

**Comment by snibgo** [comment]
Possible reasons include: the file doesn't exist, or it isn't a valid image.

If the function returns false, you can get the exception and report it to the user. See the sample program wand.c.

**Reply by programmerarunkumar** [reply]
> Possible reasons include: the file doesn't exist, or it isn't a valid image.
I have Checked , that file is exist , but how to check it is valid image or not from ImageMagick perspective 
Is there any method to validate the image ?

**Comment by programmerarunkumar** [comment]
MagickWand --> MagickReadImage(MagickWand *wand,const char *filename)  method will support multithreading ?

**Comment by snibgo** [comment]
> Is there any method to validate the image

Yes. If the function returns false, you can get the exception.

MagickReadImage() calls ReadImage() which calls the appropriate decoder function. As far as I know, all decoder functions are single-threaded.

**Reply by programmerarunkumar** [reply]
Is there Any Method  for Image Read in  Multi-threaded ?

**Comment by snibgo** [comment]
Sorry, I don't understand your question.

**Reply by programmerarunkumar** [reply]
> Sorry, I don't understand your question.

is there any method alternative to MagickReadImage() which support multi-threaded ?

**Comment by snibgo** [comment]
Perhaps you are asking: is MagickReadImage() thread-safe, meaning it can be safely called from multiple threads? I don't know.

**Comment by urban-warrior** [comment]
All methods in MagickWand are thread-safe as long as MagickWandGenesis() is the very first MagickWand method called.

**Reply by programmerarunkumar** [reply]
> All methods in MagickWand are thread-safe as long as MagickWandGenesis() is the very first MagickWand method called.

Need to call for everytime , or calling MagickWandGenesis() once at first time is enough ?

**Comment by urban-warrior** [comment]
MagickWandGenesis() is called once.  MagickWand is designed to work correctly even if its not called but it is recommended practice.
